### PR TITLE
DRAC-153 | added accounts module

### PR DESCRIPTION
**Pull request by stasmenklab**
Added accounts module

### CODE REVIEW

File: src/common/errors/errors.ts [hunk 1]
```diff
@@ -4,23 +4,65 @@ export const ERRORS = {
   auth: {
     notAuthorized: {
       code: ErrorCode.NOT_AUTHORIZED,
-      message: 'not authorized',
+      message: 'not authorized'
```
Comment (Zhuravlev1) [inline src/common/errors/errors.ts]: pls, use a prettier config from the root of project. And before pushing be sure that prettier(npm run format) was run.

File: src/modules/account/dtos/account.dto.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export class AccountDto {
+  id: number
+  name: string
+  created_at: Date
```
Comment (Zhuravlev1) [inline src/modules/account/dtos/account.dto.ts]: Pls, use cameCase format for naming variables and properties

File: src/modules/account/account.service.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+export class AccountService {
+  constructor(
+    @InjectRepository(Account)
+    private readonly accountRepository: Repository<Account>,
+    @InjectRepository(Program)
+    private readonly programRepository: Repository<Program>
+  ) {
+  }
+
+  public async getAll(program_id: number): Promise<Account[]> {
```
Comment (Zhuravlev1) [inline src/modules/account/account.service.ts]: let's change response type from **Account[]** to **AccountDto[]**
it's not a controller responsibility to build a response. So, in the service we should manipulate with object like toEntity, toDto etc.

File: src/modules/account/account.service.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  public async update(id: number, data: UpdateAccountDto): Promise<Account> {
+    const account = await this.accountRepository.findOne({
+      where: { id: id }
+    })
+
+    if (!account) {
+      throw new NotFoundException(businessException([ERRORS.account.notFound]))
+    }
+
+    toUpdateAccountEntity(account, data)
```
Comment (Zhuravlev1) [inline src/modules/account/account.service.ts]: why didn't you explicitly change the object?

File: src/modules/account/dtos/create-account/create-account.dto.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { ApiProperty } from '@nestjs/swagger'
+
+export class CreateAccountDto {
+  @ApiProperty()
+  program_id: number
```
Comment (Zhuravlev1) [inline src/modules/account/dtos/create-account/create-account.dto.ts]: pls, follow cameCase rule

File: src/modules/account/mappers/account.mapper.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { CreateAccountDto } from '../dtos/create-account/create-account.dto'
+import { UpdateAccountDto } from '../dtos/update-account/update-account.dto'
+import { Account, Program } from 'src/orm/entities'
+import { AccountDto } from '../dtos/account.dto'
+
+export function toAccountsDto(accounts: Account[]): AccountDto[] {
```
Comment (Zhuravlev1) [inline src/modules/account/mappers/account.mapper.ts]: let's create a class
```
export class AccountMapper {
 static toDto()
 static toCreateEntity()
 static toUpdateEntity()
 }
```

Also, don't need a method toAccount**s** , call **accounts.map()** in the service  level

File: src/modules/account/mappers/account.mapper.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+export function toCreateAccountEntity(program: Program, data: CreateAccountDto): Account {
+  let result = new Account()
+  result.program = program
+  result.name = data.name
+
+  return result
+}
+
+export function toUpdateAccountEntity(account: Account, data: UpdateAccountDto): Account {
+  account.name = data.name
```
Comment (Zhuravlev1) [inline src/modules/account/mappers/account.mapper.ts]: use Object.assign

File: src/modules/app/app.module.ts [hunk 8]
```diff
@@ -54,17 +60,16 @@ export class AppModule implements NestModule {
...
           secret: this.configService.get('auth.sessionSecret') as string,
-          resave: false,
+          resave: true,
           cookie: {
             sameSite: true,
-            httpOnly: false,
-            maxAge: 60000,
-          },
-        }),
+            httpOnly: false
```
Comment (Zhuravlev1) [inline src/modules/app/app.module.ts]: do we need maxAge property?

File: src/orm/entities/account.entity.ts [hunk 9]
```diff
@@ -1,9 +1,13 @@
-import { Entity, ManyToOne } from 'typeorm'
+import { Column, Entity, ManyToOne } from 'typeorm'
 import { Program } from './program.entity'
 import { BaseEntity } from './base.entity'
 
 @Entity({ name: 'account' })
 export class Account extends BaseEntity {
```
Comment (Zhuravlev1) [inline src/orm/entities/account.entity.ts]: Account -> AccountEntity

File: src/orm/entities/base.entity.ts [hunk 10]
```diff
@@ -5,8 +5,8 @@ export abstract class BaseEntity {
   id: number
 
   @CreateDateColumn()
-  createdAt: Date
+  created_at: Date
```
Comment (Zhuravlev1) [inline src/orm/entities/base.entity.ts]: revert pls

File: src/modules/account/account.controller.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+  @Post()
+  @ApiForbiddenResponse()
+  @ApiInternalServerErrorResponse()
+  @ApiNotFoundResponse(SWAGGER_OPTIONS.error)
+  @ApiBadRequestResponse(SWAGGER_OPTIONS.error)
+  public async create(@Body() createAccountDto: CreateAccountDto): Promise<AccountDto> {
+    const account = await this.accountService.create(createAccountDto)
+
+    return toAccountDto(account)
```
Comment (Zhuravlev1) [inline src/modules/account/account.controller.ts]: build a response (toDto) - it's the responsibility of service

**Review by Zhuravlev1 — CHANGES_REQUESTED** [review]

File: src/common/errors/errors.ts (hunk 1 above)
Comment (drewcouturier) [reply src/common/errors/errors.ts]: @Zhuravlev1 I can add a check for all pull requests that requires the changes to match the prettier config. Then you wouldn't have to spend time requesting formatting changes. Let me know if that would be helpful

**Review by Zhuravlev1 — APPROVED** [review]

**Pull request merged**